src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AsyncLazy`1.cs

```diff
@@ -17,6 +17,9 @@ internal static class AsyncLazy
         public static AsyncLazy<T> Create<T>(Func<CancellationToken, Task<T>> asynchronousComputeFunction)
             => new(asynchronousComputeFunction);
 
+        public static AsyncLazy<T> Create<T>(Func<CancellationToken, T> synchronousComputeFunction)
```

Comment:
Not sure if you want to call this something like CreateSynchronous just so it's a bit more clear that's the intent here, since that might be surprising for a type with "Async" in the name.

Reply:
sure!